### PR TITLE
Tutorial documentation (Part 1)

### PR DESCRIPTION
**Pull request by gordonhart**
- Quickstart (rewrite using age estimation example in this repo)
- Installing `kolena`
- Building a Workflow
- Packaging for Automated Evaluation
- FAQ (mostly a rewrite rather than a port)

Note that no effort has been made to preserve URL compatibility with the existing pages on docs.kolena.io, as most pages are not exactly 1:1.

This PR also populates various index files.

Fixes KOL-2458

### CODE REVIEW

**Comment by gordonhart** [comment]
Bumping back to draft — there are some broken links in "Packaging for Automated Evaluation" that I'd like to properly resolve in this PR, rather than just removing.

File: docs/assets/images/300-W.jpg
Comment (gordonhart) [inline docs/assets/images/300-W.jpg:1]: We're naturally going to want images inside our documentation. Any objection to keeping them inside this repo and tracking with git LFS?

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,3 @@
+docs/assets/images/CNN-DailyMail.jpg filter=lfs diff=lfs merge=lfs -text
```
Comment (yifanwu-kolena) [inline .gitattributes]: could we just track the dir like `docs/assets/images/**/*`?

Comment (gordonhart) [reply .gitattributes]: 👍 Updated

**Comment by munkyshi** [comment]
Went down a rabbithole trying to get things to run locally.
Any familiarity with this error (is this an M1 thing?):
```
OSError: no library called "cairo-2" was found
no library called "cairo" was found
no library called "libcairo-2" was found
cannot load library 'libcairo.so.2': dlopen(libcairo.so.2, 0x0002): tried: 'libcairo.so.2' (no such file), '/System/Volumes/Preboot/Cryptexes/OSlibcairo.so.2' (no such file), '/usr/lib/libcairo.so.2' (no such file, not in dyld cache), 'libcairo.so.2' (no such file), '/usr/local/lib/libcairo.so.2' (no such file), '/usr/lib/libcairo.so.2' (no such file, not in dyld cache).  Additionally, ctypes.util.find_library() did not manage to locate a library called 'libcairo.so.2'
cannot load library 'libcairo.2.dylib': dlopen(libcairo.2.dylib, 0x0002): tried: 'libcairo.2.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OSlibcairo.2.dylib' (no such file), '/usr/lib/libcairo.2.dylib' (no such file, not in dyld cache), 'libcairo.2.dylib' (no such file), '/usr/local/lib/libcairo.2.dylib' (no such file), '/usr/lib/libcairo.2.dylib' (no such file, not in dyld cache).  Additionally, ctypes.util.find_library() did not manage to locate a library called 'libcairo.2.dylib'
cannot load library 'libcairo-2.dll': dlopen(libcairo-2.dll, 0x0002): tried: 'libcairo-2.dll' (no such file), '/System/Volumes/Preboot/Cryptexes/OSlibcairo-2.dll' (no such file), '/usr/lib/libcairo-2.dll' (no such file, not in dyld cache), 'libcairo-2.dll' (no such file), '/usr/local/lib/libcairo-2.dll' (no such file), '/usr/lib/libcairo-2.dll' (no such file, not in dyld cache).  Additionally, ctypes.util.find_library() did not manage to locate a library called 'libcairo-2.dll'
```

Workaround in https://github.com/squidfunk/mkdocs-material/issues/5121 doesn't work for me either

**Comment by munkyshi** [comment]
nit: Not a fan of how the page jumps when you click a tab -- anything we can do about that:
https://github.com/kolenaIO/kolena/assets/2481598/2cd52312-a4b3-4132-8aaa-8ed9151d64af

File: docs/installing-kolena.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    poetry add kolena
+    ```
+
+!!! note "Note: Extra `metrics` dependencies"
+    Certain metrics computation functionality requires additional dependencies that can be installed via the `metrics`
+    extra:
+
+    ```shell
+    pip install kolena[metrics]
+    ```
```
Comment (munkyshi) [inline docs/installing-kolena.md]: Should this be `pip` tab specific (and should we have something for the `poetry` tab as well?)

File: docs/advanced-usage/packaging-for-automated-evaluation.md [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+automatic evaluation process.
+
+To enable automatic metrics computation when applicable, we need to package the metrics evaluation logic into a
+Docker image that the Kolena platform can run. The following sections explain how to build this Docker image and link
+it for metrics computation on the Kolena platform.
+
+## Build Evaluator Docker Image
+
+We will use the keypoints detection workflow we've built from
+[Quickstart: Building a Workflow](/testing-with-kolena/quickstart) to illustrate the process. Here
```
Comment (munkyshi) [inline docs/advanced-usage/packaging-for-automated-evaluation.md]: `/testing-with-kolena/quickstart` -> `/quickstart`?

File: docs/advanced-usage/packaging-for-automated-evaluation.md [hunk 4]
```diff
@@ -0,0 +1,348 @@
...
+│   ├── main.py
+│   └── workflow.py
+├── poetry.lock
+└── pyproject.toml
+```
+
+The `keypoints_detection` directory is where our workflow is defined, with evaluator logic in `evaluator.py` and
+workflow data objects in `workflow.py`. The `main.py` will be the entry point where `test` is executed.
+
+From the [quickstart](/testing-with-kolena/quickstart#step-4--running-tests), we know that metrics evaluation involves
```
Comment (munkyshi) [inline docs/advanced-usage/packaging-for-automated-evaluation.md]: `/quickstart/#test-a-model`?

File: docs/advanced-usage/packaging-for-automated-evaluation.md [hunk 5]
```diff
@@ -0,0 +1,348 @@
...
+docker build \
+    --tag "$IMAGE_TAG" \
+    --file "docker/Dockerfile" \
+    --build-arg KOLENA_TOKEN=${KOLENA_TOKEN} \
+    .
+```
+
+This build process installs the `kolena` package, and as such needs the `KOLENA_TOKEN` environment variable to be
+populated with your Kolena API key.
+Follow the [`kolena` Python client](testing-with-kolena/using-kolena-client#installation) guide to obtain an
```
Comment (munkyshi) [inline docs/advanced-usage/packaging-for-automated-evaluation.md]: `/installing-kolena/#initialization`?

File: docs/faq/index.md
Comment (munkyshi) [inline docs/faq/index.md:1]: nit: This much green in dark mode is a bit overwhelming:
Screenshot 2023-06-12 at 17 13 30

**Review by munkyshi — APPROVED** [review]
Read through the pages specified in the PR description:
- Quickstart (rewrite using age estimation example in this repo)
- Installing kolena
- Building a Workflow
- Packaging for Automated Evaluation
- FAQ (mostly a rewrite rather than a port)

Suggesting some fixes for links in the docs, but overall looks really good!

**Comment by gordonhart** [comment]
> nit: Not a fan of how the page jumps when you click a tab

Doesn't repro for me 🙁

File: docs/faq/index.md
Comment (gordonhart) [reply docs/faq/index.md:1]: Agreed, will look to put together a [custom admonition](https://squidfunk.github.io/mkdocs-material/reference/admonitions/#custom-admonitions) for FAQs in a follow-up.

**Comment by gordonhart** [comment]
Thanks for finding these broken links. I've fixed them on this PR, but unless we plug in an autochecker we're inevitably going to have lots of broken links. Surprisingly, there doesn't seem to be any obvious choice for such a checker — I played around with a few locally and wasn't happy with the results. Opening a ticket to track as a follow-up.

**Comment by gordonhart** [comment]
> > nit: Not a fan of how the page jumps when you click a tab
> 
> Doesn't repro for me 🙁

Seeing this now on [docs.kolena.io](https://docs.kolena.io/). Filing a ticket to track

**Pull request merged**